daemons/controld/controld_remote_ra.c

```diff
@@ -1340,6 +1357,57 @@ remote_ra_process_pseudo(xmlNode *xml)
         }
     }
     freeXpathObject(search);
+
+    /* Is this a stop pseudo-action on a remote node connection resource? */
+    search = xpath_search(xml, XPATH_PSEUDO_STOP);
+
+    if (numXpathResults(search) == 1) {
+        xmlNode *result = getXpathResult(search, 0);
+
+        if (result) {
+            const char *host = NULL;
+            const char *op_key = NULL;
+            xmlNode *attrs = NULL;
+            char *rsc_id = NULL;
+            lrm_state_t *connection_rsc = NULL;
+
+            /* Get the node that was previously hosting this remote connection
+             * resource from the pseudo-action XML
+             */
+            attrs = find_xml_node(result, XML_TAG_ATTRS, FALSE);
+            if (!attrs) {
+                goto done;
+            }
+
+            host = crm_element_value(attrs, CRM_META "_" XML_LRM_ATTR_TARGET);
+            if (!host) {
+                goto done;
+            }
+
+            /* Then get the name of the remote resource from the operation_key */
+            op_key = crm_element_value(result, XML_LRM_ATTR_TASK_KEY);
+            parse_op_key(op_key, &rsc_id, NULL, NULL);
+
+            /* Get the remote_ra_data_t associated with that resource */
+            connection_rsc = lrm_state_find(rsc_id);
+            free(rsc_id);
+
+            if (connection_rsc && connection_rsc->remote_ra_data) {
+                remote_ra_data_t *ra_data = connection_rsc->remote_ra_data;
+                crm_node_t *peer = crm_get_peer(0, host);
+
+                /* Set the timestamp on the remote data to the timestamp we got
+                 * when we first noticed the hosting peer was lost
+                 */
+                if (peer != NULL) {
+                    ra_data->when_host_lost = peer->peer_lost;
+                }
```

Comment:
It occurs to me that this will only be done on the DC (the only node that processes the transition graph). The node that will host the connection may not be the DC. I'm thinking instead of remembering this in the peer cache, we'll need to search the transition graph for the remote start, and remove the preserve flag if the uptime is too low

Reply:
Ugh, that means that we have to worry about interrupted transitions. If the transition is aborted after this point but before the remote start, the scheduler will be invoked again, and this time there won't be a pseudo-stop (which will have already completed), just the start. Then we don't have uptime information. :(

Reply:
Of course if there's no pseudo-stop, the scheduler won't add the flag to the rescheduled start. So at least it's safe, we would just be unable to preserve attributes across aborted transitions. I don't see a good way around that offhand.